build.yaml

```diff
@@ -38,3 +41,6 @@ global_options:
         BETA_SERVER_URL: https://beta.api.dartpad.dev/
         DEV_SERVER_URL: https://dev.api.dartpad.dev/
         OLD_SERVER_URL: https://old.api.dartpad.dev/
+        GH_AUTH_INIT_BASE_URL: https://githubauth-brsyns7rna-uw.a.run.app
```

Comment:
I'm curious what this Cloud Run app is, who owns it, maintenance cycle, etc

Reply:
i just pushed that to my google cloud account.  It is literally a 'gcloud run deploy --source .' of the repository, and then the CLIENT_ID and CLIENT_SECRET for the GitHub OAuth app are secrets that get injected as environmental vars, and RETURN_TO_APP_URL environmental vars inject the URL for GitHub to redirect back to the auth end point (which you register with GitHub when you register your app, and then you send again during authorization and must match), and then the AUTH_RETURN_URL var points back to https://timmaffett.github.io/index.html which is where it goes back to completing auth with github.
    I put (i think) all the details in the README.md of the repository for the auth end point, and it should just take 10 or 15 minutes to set up your own local end point (or maybe a barely little longer to deploy your own cloud endpoint). 
The source itself is really a fairly quick read and pretty simple really, it is interlaced with the docs/notes I took from the GitHub OAuth documentation.  I would of course be happy to answer any questions about it.

Reply:
I will take a look at the https://pub.dev/packages/github package - to be honest I don't think I saw that one, I did look at some package but I don't think it did the OAuth part.  
  Once authenticated I essentially extended the existing gist applications github interfacing with the additional api calls.  (the one package I had looked at did not have the gist api integration).

I will take a closer look at the dart-services, it did originally look at that before thinking for my proof of concept to just make a stand alone end point and integrate it later.

Reply:
Don't worry too much about `dart-services`, I'm happy to own integrating the required functionality into that codebase. 

@RedBrogdon and @johnpryan interested to hear your thoughts on this integration.